File: Answers/C++/1040.cpp
#include <stdio.h>

int main()
{
    double a, b, c, d, e, m;
    scanf("%lf %lf %lf %lf", &a, &b, &c, &d);
    m = (a * 2 + b * 3 + c * 4 + d) / 10;
    printf("Media: %.1f\n", m);
    if (m >= 7.0){
        printf("Aluno aprovado.\n");
    }
    else if (m >= 5.0)
    {
        printf("Aluno em exame.\n");
        scanf("%lf", &e);
        printf("Nota do exame: %.1f\n", e);
        if (e + m / 2.0 > 5.0){
            printf("Aluno aprovado.\n");
        }
        else{
            printf("Aluno reprovado.\n");
        }
        printf("Media final: %.1f\n", (e + m) / 2.0);
    }
    else{

        printf("Aluno reprovado.\n");
    }

    return 0;
//Vinicius Mangueira Correia
}

File: Answers/Python/1040.py
# -*- coding: utf-8 -*-

a, b, c, d = map(float, raw_input().split())

media = ((a*2) + (b*3) + (c*4) + (d*1)) / 10

print("Media: %.1f" % media)

if media >= 7:
    print("Aluno aprovado.")

elif media < 5:
    print("Aluno reprovado.")

else:
    print("Aluno em exame.")
    e = float(input())
    print("Nota do exame: %.1f" % e)
    exame = (media + e) / 2

    if exame >= 5:
        print("Aluno aprovado.")
        print("Media final: %.1f" % exame)
    else:
        print("Aluno reprovado.")
        print("Media final: %.1f" % exame)